paint/PAINT_Tool.cpp

```diff
@@ -16,12 +16,40 @@ void Tool::setActiveBrush(std::shared_ptr<Brush> brush)
 	activeBrush_ = std::move(brush);
 }
 
-
 void Tool::clearLines()
 {
 	lines_.clear();
 }
 
+void Tool::renderLines(SDL_Renderer* renderer, const std::vector<gfx::Line>& lines) const
+{
+	assert(activeBrush_ && "activeBrush_ is nullptr.");
+	const auto thickness = activeBrush_->getThickness();
+	assert((thickness == 0) || (thickness == 1) || (thickness == 2) && "brush thickness in renderLines is not 0, 1, or 2.");
+	for (const auto& line : lines) {
+		SDL_RenderDrawLine(renderer, line.x1, line.y1, line.x2, line.y2);
+		for (auto sign : { -1, 1 }) {
+			if (thickness > 0) {
+				for (auto thick = 0; thick <= thickness; ++thick) {
+					SDL_RenderDrawLine(renderer, line.x1 + (sign * thick), line.y1, line.x2 + (sign * thick), line.y2);
+					SDL_RenderDrawLine(renderer, line.x1, line.y1 + (sign * thick), line.x2, line.y2 + (sign * thick));
+				}
+			}
+
+			if (thickness > 1) {
+
+				SDL_RenderDrawLine(renderer, line.x1 + (sign * 1), line.y1 + (sign * 1), line.x2 + (sign * 1), line.y2 + (sign * 1));
```

Comment:
Consider using a filled polygon rather than a line for thickness > 1. Measure the speed gain. If it's better, keep it. If not, revert.
**COULD**

Reply:
SDL doesnt have this option - would have to use another external library. could create a ticket for it.